app/qmlV2/component/MMAbstractEditor.qml

```diff
@@ -0,0 +1,181 @@
+
+      Rectangle {
+        id: background
+
+        width: parent.width
+        height: parent.height
+        border.color: root.hasFocus ? ( errorMsg.length > 0 ? StyleV2.negativeColor : warningMsg.length > 0 ? StyleV2.warningColor : StyleV2.forestColor ) : StyleV2.transparentColor
+        border.width: 2 * __dp
+        color: root.bgColor
+        radius: 12 * __dp
+      }
+
+      Row {
+        height: parent.height
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.leftMargin: root.spacing
+        anchors.rightMargin: root.spacing
+
+        Item {
+          id: leftActionContainer
+
+          property bool actionAllowed: leftActionContainer.children.length > 1 && (leftActionContainer.children[1]?.visible ?? false)
+
+          height: parent.height
+          width: actionAllowed ? height/2 : 0
+
+          Rectangle {
+            color: StyleV2.transparentColor
+            anchors.centerIn: parent
+            width: leftActionContainer.actionAllowed ? parent.width + root.spacing/2 : 0
+            height: parent.height
+
+            MouseArea {
+              anchors.fill: parent
+              onReleased: root.leftActionClicked()
+            }
+          }
+        }
+
+        Item {
+          id: contentContainer
+
+          height: parent.height
+          width: parent.width - (leftActionContainer.actionAllowed ? leftActionContainer.width : 0) - (rightActionContainer.actionAllowed ? rightActionContainer.width : 0) //- 2 * root.spacing - ( leftActionContainer.actionAllowed ? root.spacing : 0 )
+
+          MouseArea {
+            anchors.fill: parent
+
+            onClicked: {
+              root.contentClicked()
+            }
+          }
+        }
+
+        Item {
+          id: rightActionContainer
+
+          property bool actionAllowed: rightActionContainer.children.length > 1 && (rightActionContainer.children[1]?.visible ?? false)
+
+          height: parent.height
+          width: actionAllowed ? height/2 : 0
+
+          Rectangle {
+            color: StyleV2.transparentColor
+            anchors.centerIn: parent
+            width: rightActionContainer.actionAllowed ? parent.width + root.spacing/2 : 0
+            height: parent.height
+
+            MouseArea {
+              anchors.fill: parent
+              onReleased: root.rightActionClicked()
+            }
+          }
+        }
+      }
+    }
+
+    Item {
+      id: messageItem
+
+      width: parent.width
+      height: msgRow.height
+
+      Row {
+        id: msgRow
+
+        spacing: 4 * __dp
+
+        MMIcon {
+          id: msgIcon
+
+          source: visible ? StyleV2.errorIcon : ""
+          color: errorMsg.length > 0 ? StyleV2.negativeColor : StyleV2.warningColor
+          visible: errorMsg.length > 0 || warningMsg.length > 0
+        }
+        Text {
+          text: root.errorMsg.length > 0 ? root.errorMsg : root.warningMsg
+          font: StyleV2.t4
```

Comment:
We need to limit the maximum number of lines to be 2 https://doc.qt.io/qt-6/qml-qtquick-text.html#maximumLineCount-prop